packages/web-frontend/src/components/View/DownloadFilesVisual.js

```diff
@@ -19,22 +20,34 @@ const StyledDownloadFilesVisual = styled(BaseDownloadFilesVisual)`
   }
 `;
 
-const getFilenameFromPath = fileName => fileName.split('/').at(-1);
-const getZipFileName = () =>
-  `tupaia-download-${new Date().toLocaleString('default', {
-    year: 'numeric',
-    month: '2-digit',
-    day: '2-digit',
-  })}.zip`;
-
 export const DownloadFilesVisual = ({ onClose: originalOnClose, ...restOfProps }) => {
   const [error, setError] = useState(null);
 
-  const downloadFiles = async files => {
+  const downloadFiles = async uniqueFileNames => {
     try {
-      const filesString = files.join(',');
-      const fileName = files.length === 1 ? getFilenameFromPath(files[0]) : getZipFileName();
-      await download(`downloadFiles?files=${filesString}`, undefined, {}, fileName);
+      if (uniqueFileNames.length === 1) {
+        const { fileName } = getUniqueFileNameParts(uniqueFileNames[0]);
+        await download(
+          `downloadFiles?uniqueFileNames=${uniqueFileNames[0]}`,
+          undefined,
+          {},
+          fileName,
+        );
```

Comment:
Ideally we wouldn't have to do this, and we could just rely on the filename that came back from the server. I'd kind of prefer that since it means this viz is agnostic to the whole `uniqueFilename` concept. It does come back in the `content-disposition` header I think. But at the time I didn't see an easy way to access it.

Can I ask you to take a quick look to see if it's possible?

Also, if it is possible, then I think we can do the same for the zip, if we pass in the `zipFileName` query param

Reply:
Yeah that confused me a lot when I came across it. I decided to leave it as is to keep changes to the frontend small for the rewrite. Definitely something that would be nice to clean up in the future though.

Reply:
I suspect we'd use a different method (and probably different library) to do this post frontend re-write, so yeah happy to hold off til then as well